src/embedded/index.ts

```diff
@@ -145,80 +151,173 @@ export class EmbeddedDB {
-  const tarballPath = `${opt.binaryPath}.tgz`;
-  const file = fs.createWriteStream(tarballPath);
-  return new Promise((resolve, reject) => {
-    const url =
-      'https://github.com/weaviate/weaviate/releases' +
-      `/download/v${opt.version}/weaviate-v${opt.version}-linux-amd64.tar.gz`;
-    get(url, (resp) => {
-      if (resp.statusCode == 302 && resp.headers.location) {
-        get(resp.headers.location, (resp) => {
-          resp.pipe(file);
-          file.on('finish', () => {
-            file.close();
-            resolve(tarballPath);
+  private downloadBinary(): Promise<string> {
+    const tarballPath = `${this.options.binaryPath}.tgz`;
+    const file = fs.createWriteStream(tarballPath);
+    return new Promise((resolve, reject) => {
+      const url = this.buildBinaryUrl();
+      get(url, (resp) => {
+        if (resp.statusCode == 302 && resp.headers.location) {
+          get(resp.headers.location, (resp) => {
+            resp.pipe(file);
+            file.on('finish', () => {
+              file.close();
+              resolve(tarballPath);
+            });
           });
-        });
-      } else if (resp.statusCode == 404) {
-        reject(
-          new Error(
-            `failed to download binary: not found. ` +
-              `are you sure Weaviate version ${opt.version} exists? ` +
-              `note that embedded db is only supported for versions >= 1.18.0`
-          )
-        );
-      } else {
-        reject(new Error(`failed to download binary: unexpected status code: ${resp.statusCode}`));
-      }
-    }).on('error', function (err) {
-      fs.unlinkSync(tarballPath);
-      reject(new Error(`failed to download binary: ${JSON.stringify(err)}`));
+        } else if (resp.statusCode == 404) {
+          reject(
+            new Error(
+              `failed to download binary: not found. ` +
+                `are you sure Weaviate version ${this.options.version} exists? ` +
+                `note that embedded db is only supported for versions >= 1.18.0`
+            )
+          );
+        } else {
+          reject(new Error(`failed to download binary: unexpected status code: ${resp.statusCode}`));
+        }
+      }).on('error', (err) => {
+        fs.unlinkSync(tarballPath);
+        reject(new Error(`failed to download binary: ${JSON.stringify(err)}`));
+      });
     });
-  });
-}
+  }
 
-function untarBinary(opt: EmbeddedOptions, tarballPath: string): Promise<null> {
-  const tarball = fs.createReadStream(tarballPath);
-  return new Promise((resolve, reject) => {
-    tarball.pipe(
-      extract({
-        cwd: dirname(tarballPath),
-        strict: true,
-      })
-        .on('finish', () => {
-          tarball.close();
-          fs.renameSync(join(dirname(opt.binaryPath), 'weaviate'), opt.binaryPath);
-          resolve(null);
+  private buildBinaryUrl(): string {
+    let arch: string;
+    switch (process.arch) {
+      case 'arm64':
```

Comment:
did you test this?
On python the platform command returns aarch64 (which is taken from uname), no idea what the TS library does

(Afaik we do not have anyone with an ARM64 laptop)

Reply:
Yes, I have tested on both arm64 and amd64 environments.

Also see: https://nodejs.org/api/process.html#processarch